### PR TITLE
Move Container Apps into Terraform

### PR DESCRIPTION
**Pull request by nickclyde**
Resolves https://app.clickup.com/t/384h6ca

Not ready for testing until `3.43.0` of the terraform azure provider is released

### CODE REVIEW

**Comment by nickclyde** [comment]
Just confirmed this works by doing a fresh deploy to `dev4` and successfully running the pipeline! Ready for review!

File: .github/workflows/deployment.yaml [hunk 1]
```diff
@@ -95,149 +94,6 @@ jobs:
         run: |
           terraform init -backend-config=backend.tfvars
           terraform workspace new $TF_ENV || terraform workspace select $TF_ENV
-          terraform apply -target="module.shared.azurerm_container_registry.phdi_registry" -target="module.shared.azurerm_user_assigned_identity.pipeline_runner" -target="module.shared.azurerm_storage_account.phi" -target="module.shared.azurerm_storage_share.tables" -target="module.shared.azurerm_key_vault.phdi_key_vault" -target="module.shared.random_uuid.salt" -target="module.shared.azurerm_key_vault_secret.salt" -target="module.shared.docker_image.ghcr_image" -target="module.shared.docker_tag.tag_for_azure" -target="module.shared.docker_registry_image.acr_image" -auto-approve -lock-timeout=30m
```
Comment (DanPaseltiner) [inline .github/workflows/deployment.yaml:98]: This is one of the most beautiful diffs I have ever seen. Good riddance!

File: terraform/implementation/backend.tf [hunk 2]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "= 3.23.0"
+      version = "= 3.43.0"
```
Comment (DanPaseltiner) [inline terraform/implementation/backend.tf:5]: WooHoo!

**Review by DanPaseltiner — APPROVED** [review]
LGTM! Thank you for making this happen @nickclyde. Major improvement for us.

**Pull request merged**